test/src/util/pelz_json_parser_test_suite.c

```diff
@@ -0,0 +1,440 @@
+  int dec_data_len = 57;
+
+  //Building of a standard valid JSON request
+  json = cJSON_CreateObject();
+  cJSON_AddItemToObject(json, "request_type", cJSON_CreateNumber(1));
+  cJSON_AddItemToObject(json, "key_id", cJSON_CreateString(json_key_id));
+  cJSON_AddItemToObject(json, "key_id_len", cJSON_CreateNumber(json_key_id_len));
+  cJSON_AddItemToObject(json, "enc_data", cJSON_CreateString(enc_data));
+  cJSON_AddItemToObject(json, "enc_data_len", cJSON_CreateNumber(enc_data_len));
+  cJSON_AddItemToObject(json, "dec_data", cJSON_CreateString(dec_data));
+  cJSON_AddItemToObject(json, "dec_data_len", cJSON_CreateNumber(dec_data_len));
+
+  //Test standard valid JSON request
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 0);
+  freeCharBuf(&key_id);
+  freeCharBuf(&data);
+
+  //Test check of JSON request hasObject
+  cJSON_DeleteItemFromObject(json, "key_id");
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 1);
+  cJSON_AddItemToObject(json, "key_id", cJSON_CreateString(json_key_id));
+
+  cJSON_DeleteItemFromObject(json, "key_id_len");
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 1);
+  cJSON_AddItemToObject(json, "key_id_len", cJSON_CreateNumber(json_key_id_len));
+
+  cJSON_DeleteItemFromObject(json, "enc_data");
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 1);
+  cJSON_AddItemToObject(json, "enc_data", cJSON_CreateString(enc_data));
+
+  cJSON_DeleteItemFromObject(json, "enc_data_len");
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 1);
+  cJSON_AddItemToObject(json, "enc_data_len", cJSON_CreateNumber(enc_data_len));
+
+  cJSON_DeleteItemFromObject(json, "dec_data");
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 0);
+  freeCharBuf(&key_id);
+  freeCharBuf(&data);
+  cJSON_AddItemToObject(json, "dec_data", cJSON_CreateString(dec_data));
+
+  cJSON_DeleteItemFromObject(json, "dec_data_len");
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 0);
+  freeCharBuf(&key_id);
+  freeCharBuf(&data);
+  cJSON_AddItemToObject(json, "dec_data_len", cJSON_CreateNumber(dec_data_len));
+
+  //Test check of JSON request isNumber
+  cJSON_DeleteItemFromObject(json, "key_id_len");
+  cJSON_AddItemToObject(json, "key_id_len", cJSON_CreateString("19"));
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 1);
+  cJSON_DeleteItemFromObject(json, "key_id_len");
+  cJSON_AddItemToObject(json, "key_id_len", cJSON_CreateNumber(json_key_id_len));
+
+  cJSON_DeleteItemFromObject(json, "enc_data_len");
+  cJSON_AddItemToObject(json, "enc_data_len", cJSON_CreateString("45"));
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 1);
+  cJSON_DeleteItemFromObject(json, "enc_data_len");
+  cJSON_AddItemToObject(json, "enc_data_len", cJSON_CreateNumber(enc_data_len));
+
+  cJSON_DeleteItemFromObject(json, "dec_data_len");
+  cJSON_AddItemToObject(json, "dec_data_len", cJSON_CreateString("57"));
+  CU_ASSERT(encrypt_parser(json, &key_id, &data) == 0);
```

Comment:
Why would `encrypt_parser` ever get handed a JSON object with `dec_data_len`? If the correct behavior is to ignore that field anyway I'm not sure this test needs to be here.